### PR TITLE
native coverage: reflections registration of constructors

### PR DESCRIPTION
**Pull request by pjgg**
original issue Ref: https://github.com/quarkusio/quarkus/issues/19711

A new issue was created: https://github.com/quarkusio/quarkus/issues/22654

### CODE REVIEW

File: http/jaxrs/src/test/java/io/quarkus/ts/security/core/SerializationIT.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public void serializeList() {
+        String result = makeQuery("/hello/serialize-list").asString();
+        Assertions.assertEquals("[Hello RESTEasy]", result);
+    }
+
+    @Test
+    @Disabled("https://github.com/quarkusio/quarkus/issues/22654")
+    public void serializeBigList() {
+        String basePath = "/hello/big-serialize-list?expSize=";
+        mustBeGreaterThan(makeQuery(basePath + (LONGEST_EAGER_ALLOC - 39)).jsonPath().getInt("size"), 0);
```
Comment (fedinskiy) [inline http/jaxrs/src/test/java/io/quarkus/ts/security/core/SerializationIT.java:33]: Too many magic numbers in the method. Probably, these should be separate constants

File: http/jaxrs/src/main/java/io/quarkus/ts/security/core/GreetingResource.java [hunk 2]
```diff
@@ -1,17 +1,127 @@
...
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("/serialize-list")
+    public String serlist() throws IOException, ClassNotFoundException {
+
+        byte[] bytes;
+
+        try (
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
```
Comment (fedinskiy) [inline http/jaxrs/src/main/java/io/quarkus/ts/security/core/GreetingResource.java:64]: Can we use jdkSerialize and jdkDeserialize in all methods of this class?

**Review by fedinskiy — CHANGES_REQUESTED** [review]

File: http/jaxrs/src/main/java/io/quarkus/ts/security/core/GreetingResource.java (hunk 2 above)
Comment (pjgg) [reply http/jaxrs/src/main/java/io/quarkus/ts/security/core/GreetingResource.java:64]: We have an issue with the native mode that is under investigation. Have a look at the description of this ticket. I think that maybe in the future `jdkSerialize` and `jdkDeserialize` are going to be removed.

File: http/jaxrs/src/test/java/io/quarkus/ts/security/core/SerializationIT.java (hunk 1 above)
Comment (pjgg) [reply http/jaxrs/src/test/java/io/quarkus/ts/security/core/SerializationIT.java]: These "magic numbers" are random variations just to tease out possible edge cases. 
The original scenario is the following one: https://github.com/FasterXML/jackson-databind/commit/3ccde7d938fea547e598fdefe9a82cff37fed5cb

**Review by fedinskiy — APPROVED** [review]
